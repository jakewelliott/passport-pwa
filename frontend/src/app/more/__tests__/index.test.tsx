import More from '@/app/more/index';
import { renderWithClient } from '@/lib/testing/test-wrapper';
import { screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

describe('More Page', () => {
  it('renders all links correctly', () => {
    renderWithClient(<More />);

    // Check for all links
    const links = [
      { text: 'Trails', href: '/more/trails' },
      { text: 'Bucket List', href: '/more/bucket-list' },
      { text: 'My Notes', href: '/more/my-notes' },
      { text: 'Welcome Message', href: '/more/welcome-message' },
      { text: 'Staying Safe', href: '/more/staying-safe' },
      { text: 'Hiking Essentials', href: '/more/hiking-essentials' },
      { text: 'Icon Legend', href: '/more/icon-legend' },
      { text: 'App Info', href: '/more/app-info' },
    ];

    for (const { text, href } of links) {
      const linkElement = screen.getByText(text).closest('a');
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveAttribute('href', href);
    }
  });

  // TOOD: we need to get rid of user store
  // it("displays the logged-in user's username", () => {
  // 	renderWithClient(<More />);
  // 	expect(screen.getByText("You are currently logged in as 'testUser'")).toBeInTheDocument();
  // });
});
