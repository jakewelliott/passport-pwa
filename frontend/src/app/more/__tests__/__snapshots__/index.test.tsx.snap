// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`More Page > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-3.5"
    data-testid="more-list"
  >
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/trails"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          Trails
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/bucket-list"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          Bucket List
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/my-notes"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          My Notes
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/welcome-message"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          Welcome Message
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/staying-safe"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          Staying Safe
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/hiking-essentials"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          Hiking Essentials
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/icon-legend"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          Icon Legend
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/ncdpr-info"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          NCDPR Info
        </h2>
      </div>
    </a>
    <a
      class="text-supporting-inactiveblue no-underline"
      data-discover="true"
      href="/more/my-profile"
    >
      <div
        class="flex items-center bg-supporting-lightblue p-3 shadow-md"
        style="border-radius: 15px;"
      >
        <h2
          class="mx-4 my-6"
        >
          My Profile
        </h2>
      </div>
    </a>
  </div>
</div>
`;
